lib/MusicBrainz/Server/WebService/XMLSerializer.pm

```diff
@@ -228,9 +228,10 @@ sub _serialize_collection
 
     my $ser = "_serialize_${entity_type}_list";
     my $gen_list = "${entity_type}_list";
+    my $list = $opts->{$plural};
 
-    if ($toplevel) {
-        $self->$ser(\@collection, $gen, $opts->{$plural}, $inc, $stash);
+    if ($toplevel && defined($list->{items}) && @{ $list->{items} }) {
```

Comment:
It's basically checking that the array is not empty, with some Perlisms: `$list->{items}` is an ARRAY reference, `@{ $list->{items} }` dereferences it into an array, and inside the `if` it's evaluated in scalar context which returns the length of the array.